sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/EncryptionCosmosAsyncContainer.java

```diff
@@ -242,22 +259,22 @@
             Preconditions.checkArgument(partitionKey != null, "partitionKey cannot be null for operations using "
                 + "EncryptionContainer.");
 
-            byte[] payload = cosmosSerializerToStream(item);
-            Mono<CosmosItemResponse<byte[]>> result = this.replaceItemStream(payload,
-                itemId,
-                partitionKey,
-                encryptionItemRequestOptions);
+            return Mono.defer(() -> {
+                byte[] payload = cosmosSerializerToStream(item);
+                Mono<CosmosItemResponse<byte[]>> result = this.replaceItemStream(payload,
+                    itemId,
+                    partitionKey,
+                    encryptionItemRequestOptions);
 
-            return result.map(rsp -> (CosmosItemResponse<T>) this.responseFactory.createItemResponse(rsp,
-                item.getClass()));
+                return result.map(rsp -> (CosmosItemResponse<T>) this.responseFactory.createItemResponse(rsp,
+                    item.getClass()));
+            }).subscribeOn(encryptionScheduler);
```

Comment:
`SubsribeOn` will also change the actual call - which in this case could be all the way back to azure-cosmos. Any reason why you are not using `PublishOn` here ?

Reply:
decryption will happen when Flux/Mono emits results (after reading data from service) and so we need `publishOn` for decryption to ensure it runs on the desired scheduler.

encryption will happen prior to sending data to service and we have a `defer` operator which in conjunction of `subscribeOn` will ensure the actual encryption call to happen on the desired scheduler.